gcp/appengine/generate_sitemap.py

```diff
@@ -0,0 +1,141 @@
+  ecosystem_name = ecosystem.replace(' ', '_').replace('.', '__').strip()
+  return f'{_SITEMAPS_DIRECTORY}/{ecosystem_name}.xml'
+
+
+def get_sitemap_url_for_ecosystem(ecosystem: str, base_url: str) -> str:
+  ecosystem_name = ecosystem.replace(' ', '_').replace('.', '__').strip()
+  return f'{base_url}/sitemap/{ecosystem_name}.xml'
+
+
+def generate_sitemap_for_ecosystem(ecosystem: str, base_url: str) -> None:
+  """Generate a sitemap for the give n ecosystem."""
+  os.makedirs(_SITEMAPS_DIRECTORY, exist_ok=True)
+
+  vulnerability_ids = fetch_vulnerability_ids(ecosystem)
+  filename = get_sitemap_filename_for_ecosystem(ecosystem)
+  urlset = Element(
+      'urlset', xmlns="http://www.sitemaps.org/schemas/sitemap/0.9")
+
+  # TODO: For large ecosystems with over 50,000 vulnerabilities, generate
+  # multiple sitemaps.
+  for vuln in vulnerability_ids[:_SITEMAP_URL_LIMIT]:
+    url = SubElement(urlset, 'url')
+    loc = SubElement(url, 'loc')
+    loc.text = f"{base_url}/vulnerability/{vuln}"
+    lastmod = SubElement(url, 'lastmod')
+    lastmod.text = datetime.datetime.now().isoformat()
+
+  tree = ElementTree(urlset)
+  tree.write(filename, encoding='utf-8', xml_declaration=True)
+
+
+def compress_file(file_path: str) -> str:
+  """Compress the file using gzip and return the path to the compressed file."""
+  base, _ = os.path.splitext(file_path)
+  compressed_file_path = f"{base}.gz"
+  with open(file_path, 'rb') as f_in:
+    with gzip.open(compressed_file_path, 'wb') as f_out:
+      shutil.copyfileobj(f_in, f_out)
+  # Remove the original uncompressed file
+  os.remove(file_path)
+  return compressed_file_path
+
+
+def generate_sitemap_index(ecosystems: set[str], base_url: str) -> None:
+  """Generate a sitemap index."""
+  os.makedirs(_SITEMAPS_DIRECTORY, exist_ok=True)
+
+  sitemapindex = Element(
+      'sitemapindex', xmlns="http://www.sitemaps.org/schemas/sitemap/0.9")
+
+  for ecosystem in ecosystems:
+    sitemap = SubElement(sitemapindex, "sitemap")
+    loc = SubElement(sitemap, 'loc')
+    loc.text = compress_file(get_sitemap_url_for_ecosystem(ecosystem, base_url))
+    lastmod = SubElement(sitemap, 'lastmod')
+    lastmod.text = datetime.datetime.now().isoformat()
+
+  tree = ElementTree(sitemapindex)
+  tree.write(_SITEMAP_INDEX_PATH, encoding='utf-8', xml_declaration=True)
+  compress_file(_SITEMAP_INDEX_PATH)
+
+
+def generate_sitemaps(base_url: str) -> None:
+  """Generate sitemaps including all vulnerabilities, split by ecosystem."""
+
+  # Go over the base ecosystems index. Otherwise we'll have duplicated
+  # vulnerabilities in the sitemap.
+  base_ecosystems = {
+      ecosystem for ecosystem in osv_get_ecosystems() if ':' not in ecosystem
+  }
+  for ecosystem in base_ecosystems:
+    generate_sitemap_for_ecosystem(ecosystem, base_url)
+    compress_file(get_sitemap_filename_for_ecosystem(ecosystem))
+
+  generate_sitemap_index(base_ecosystems, base_url)
+  compress_file(_SITEMAP_INDEX_PATH)
```

Comment:
The index file is already compressed and deleted on line 109

Reply:
Yes was an unwanted consequence of a rebase. It should be fixed now.